cmd/gke-gcloud-auth-plugin/main.go

```diff
@@ -80,26 +111,38 @@ func execCredential() (*clientauth.ExecCredential, error) {
 	}, nil
 }
 
-func accessToken() (string, *meta.Time, error) {
-	if !*useAdcPtr {
-		token, expiry, err := gcloudAccessToken()
-		if err == nil {
-			return token, expiry, nil
-		}
+func (p *plugin) accessToken() (string, *metav1.Time, error) {
+	if !p.useApplicationDefaultCredentials {
+		return p.gcloudAccessToken()
 	}
-	return defaultAccessToken()
+	return p.defaultAccessToken()
 }
 
-func gcloudAccessToken() (string, *meta.Time, error) {
-	gc, err := retrieveGcloudConfig()
+func (p *plugin) gcloudAccessToken() (string, *metav1.Time, error) {
+	gc, err := p.readGcloudConfig()
 	if err != nil {
 		return "", nil, err
 	}
+	if gc.Credential.AccessToken == "" {
```

Comment:
Will an access token always be returned, even if authorization_token_file is set?

Reply:
Yes. We need both.